network/src/constants.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+/// A collection of constants and default values for configuring various network components.
+
+// NB: Almost all of these values are educated guesses, and not determined using any empirical
+// data. If you run into a limit and believe that it is unreasonably tight, please submit a PR
+// with your use-case. If you do change a value, please add a comment linking to the PR which
+// advocated the change.
+// TODO:  Each of these should be commented with semantic meaning, intended use place, and justification for the value.
+// TODO:  Better --- these should be encapsulated in configurations somewhere.
```

Comment:
my first thought...

Reply:
These constants should absolutely *NOT* be constants and need to be provided through configuration so the values can be operationally changed.  The PRs coming through in the last week to change these for production reasons are red flags.

This is a problem that exists in the code today;  a side effect of this PR is to make the existence of that problem much more visible and explicit.  Fixing it is orthogonal to reworking node setup/XXX_builder and can be done (a) on the side or (b) after the fact;  it should not be inserted as a blocking step to cleaning the builder mechanics.  In general, I believe that the builder refactoring will make it very clear which configuration should hold each of these values;  so I vote for (b).